core/src/main/java/io/grpc/HandlerRegistry.java

```diff
@@ -42,6 +43,12 @@
 public abstract class HandlerRegistry {
 
   /**
+   * Returns the {@link ServerServiceDefinition}s provided by the registry.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
+  public abstract Collection<ServerServiceDefinition> getServices();
```

Comment:
FYI this will break existing concrete implementations since they won't have this method.

Are usual work around has been to implement the method, but make it throw UnsupportedOperationException